backend/dataall/modules/dataset_sharing/db/share_object_repositories.py

```diff
@@ -288,14 +288,18 @@ def update_state(self, session, share_uri, new_state):
             return True
 
     def update_state_single_item(self, session, share_item, new_state):
-        logger.info(f"Updating share item in DB {share_item.shareItemUri} status to {new_state}")
-        ShareObjectRepository.update_share_item_status(
-            session=session,
-            uri=share_item.shareItemUri,
-            status=new_state
-        )
-        self._state = new_state
-        return True
+        try:
+            logger.info(f"Updating share item in DB {share_item.shareItemUri} status to {new_state}")
+            ShareObjectRepository.update_share_item_status(
```

Comment:
Imo returning FALSE here is not OK because it is a critical failure which cannot and should not be ignored or swallowed.. Because this line is the line that marks in-progress to FAILED... So I would change this to:

logger.error("Could not update share item status: " , exc_info=True)
raise e

Reply:
I agree

Reply:
we can resolve